src/components/Missions.js

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  setMissions,
+  joinMission,
+  leaveMission,
+} from '../redux/missions/missionSlice';
+import './styles/Missions.css';
+
+function Missions() {
+  const dispatch = useDispatch();
+  const missions = useSelector((state) => state.mission.missions);
+  const joinedMissions = useSelector((state) => state.mission.joinedMissions);
+
+  useEffect(() => {
+    const fetchMissions = async () => {
+      try {
+        const response = await fetch('https://api.spacexdata.com/v3/missions');
+        const data = await response.json();
+        dispatch(setMissions(data));
+      } catch (error) {
+        console.error('Error fetching missions:', error);
+      }
+    };
+
+    fetchMissions();
```

Comment:
- [ ] I think components should simply wait around to handle data from redux where the app state is being managed. Hence this logic should have been done as in Rockets in rocketSlice. Components should only call the API module directly and then dispatch the action.

Reply:
Implemented those changes